### PR TITLE
Allow input fields to be rendered as readonly

### PR DESCRIPTION
**Pull request by dharrigan**
Hi,

Allow input fields to be rendered as readonly. This helps when displaying data in a form (to adhere to the form layout of the other input elements), but clearly mark it as readonly so modifications are not allowed.

-=david=-

### CODE REVIEW

**Comment by AdSchellevis** [comment]
Hi David, 

Wouldn't the same also count for the other input types? Adding this to the templates is no problem, but maybe we should extend it to the input tags as well....

Best regards,

Ad

**Comment by dharrigan** [comment]
Hi,

Absolutely, but for now, I only (selfish reasons :-)) have need for it on the textarea input element. I would be happy to add it to others in another PR in a little while.

-=david=-

**Comment by AdSchellevis** [comment]
yes please :)

**Comment by dharrigan** [comment]
Could you merge in this one first please, if acceptable? I sort of need it for my scp-backup plugin (to render the public key on the page, but as readonly :-))

File: src/opnsense/mvc/app/views/layout_partials/form_input_tr.volt [hunk 1]
```diff
@@ -65,7 +66,7 @@ allownew    :   allow new items (for list) if applicable
         {% elseif type == "password" %}
             <input type="password" class="form-control" size="{{size|default("50")}}" id="{{ id }}" >
         {% elseif type == "textbox" %}
-            <textarea rows="{{height|default("5")}}" id="{{ id }}"></textarea>
+            <textarea rows="{{height|default("5")}}" id="{{ id }}" {{ readonly ? 'readonly' : '' }}></textarea>
```
Comment (fichtner) [inline src/opnsense/mvc/app/views/layout_partials/form_input_tr.volt]: readonly="readonly"

is XHTML compatible

File: src/opnsense/mvc/app/views/layout_partials/form_input_tr.volt [hunk 2]
```diff
@@ -38,6 +38,7 @@ style       :   css class to add
 maxheight   :   maximum height in rows if applicable
 width       :   width in pixels if applicable
 allownew    :   allow new items (for list) if applicable
+readonly    :   if true, then textareas (textboxes) will be readonly
```
Comment (fichtner) [inline src/opnsense/mvc/app/views/layout_partials/form_input_tr.volt]: better to omit "(textboxes)" if we want to fix all

Comment (fichtner) [reply src/opnsense/mvc/app/views/layout_partials/form_input_tr.volt]: PS: read-only

Comment (fichtner) [reply src/opnsense/mvc/app/views/layout_partials/form_input_tr.volt]: (in the description)

**Pull request closed without merging**